src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -39,6 +39,16 @@ private sealed partial class UIThreadOnly : IDisposable
 
             private readonly IInteractiveWindowEditorFactoryService _factory;
 
+            private readonly ITextBufferUndoManagerProvider _textBufferUndoManagerProvider;
+            private ITextBufferUndoManager UndoManager { set; get; }
+            public ITextUndoHistory UndoHistory
```

Comment:
You are right about `UIThreadOnly`, it's basically a collection of interactive window operations that must be ran on UI thread. The only reason UndoHistory is exposed is for unit test purpose (so we can redo/undo in test cases), the corresponding methods with thread checks are added in `InteractiveWindow.cs`. I can make it private and create `Undo` `Redo` helpers here, but I don't see a big difference though. Maybe a comment says this is exposed for test purpose would be enough?